src/contract/assembled_transaction.ts

```diff
@@ -741,4 +804,64 @@ export class AssembledTransaction<T> {
       .readWrite().length;
     return authsCount === 0 && writeLength === 0;
   }
+
+  /**
+   * Restores the footprint (resource ledger entries that can be read or written) 
+   * of an expired transaction. 
+   * 
+   * The method will:
+   * 1. Build a new transaction aimed at restoring the necessary resources.
+   * 2. Sign this new transaction if a `signTransaction` handler is provided.
+   * 3. Send the signed transaction to the network.
+   * 4. Await and return the response from the network.
+   * 
+   * Preconditions:
+   * - A `signTransaction` function must be provided during the Client initialization.
+   * - The provided `restorePreamble` should include a minimum resource fee and valid
+   *   transaction data.
+   * 
+   * @throws {Error} - Throws an error if no `signTransaction` function is provided during 
+   * Client initialization.
+   * @throws {AssembledTransaction.Errors.RestoreFailure} - Throws a custom error if the 
+   * restore transaction fails, providing the details of the failure.
+ */
+  async restoreFootprint(
+    /**
+     * The preamble object containing data required to 
+     * build the restore transaction.
+     */
+    restorePreamble: {
+      minResourceFee: string;
+      transactionData: SorobanDataBuilder;
+    },
+    /**
+     * The account that is executing the footprint restore operation.
+     */
```

Comment:
I prefer one-line comments to use the inline jsdoc syntax for brevity :pray: 
```suggestion
    /** The account that is executing the footprint restore operation. */
```